src/ArmTemplates/Extractor/Utilities/DataProcessors/ApiDataProcessor.cs

```diff
@@ -0,0 +1,44 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Apis;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors.Absctraction;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors
+{
+    public class ApiDataProcessor : IApiDataProcessor
+    {
+        public void ProcessData(List<ApiTemplateResource> apis)
+        {
+            if (apis.IsNullOrEmpty())
+            {
+                return;
+            }
+
+            foreach (var api in apis)
+            {
+                this.ProcessSingleData(api);
+            }
+        }
+
+        public void ProcessSingleData(ApiTemplateResource api)
+        {
+            api.OriginalName = api.Name;
+            api.Properties.LocalIsCurrent = api.Properties.IsCurrent;
+            api.Properties.IsCurrent = null;
```

Comment:
can we supply a proper comment on why we explicitly want to hide a `isCurrent`.

also - do we really need a `localIsCurrent`? First of all naming doesn't explain the field to me, and I dont see it is being used anywhere else in other classes. Can it be just a local variable here?

Reply:
Yes, I was using this localIsCurrent variable in first versions of the PR, which were then deprecated after redesigning. 
Regarding removing isCurrent at all: 
  When creating the revision from template, it is not possible to set the created revision as current within api contract, it will raise the exception. To make it current release should be deployed.